2018-12-30 15:36:48,656 ERROR: Exception on /index [GET] [in c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\app.py:1761]
Traceback (most recent call last):
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\_compat.py", line 35, in reraise
    raise value
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\Simon Okello\Documents\ANDELA\Flask\app\routes.py", line 45, in index
    return render_template("index.html", title="Home", form=form, posts=posts)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\templating.py", line 135, in render_template
    context, ctx.app)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\templating.py", line 117, in _render
    rv = template.render(context)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\Simon Okello\Documents\ANDELA\Flask\app\templates\index.html", line 1, in top-level template code
    {% extends "layout.html" %}
  File "C:\Users\Simon Okello\Documents\ANDELA\Flask\app\templates\layout.html", line 33, in top-level template code
    {% block content %}
  File "C:\Users\Simon Okello\Documents\ANDELA\Flask\app\templates\index.html", line 6, in block "content"
    {{form.hidden_tag()}}
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'form' is undefined
2018-12-30 15:36:56,791 INFO: Flask_Demo startup [in C:\Users\Simon Okello\Documents\ANDELA\Flask\app\__init__.py:45]
2018-12-30 15:38:11,017 INFO: Flask_Demo startup [in C:\Users\Simon Okello\Documents\ANDELA\Flask\app\__init__.py:45]
2018-12-30 15:44:03,484 INFO: Flask_Demo startup [in C:\Users\Simon Okello\Documents\ANDELA\Flask\app\__init__.py:45]
2018-12-30 15:53:56,929 INFO: Flask_Demo startup [in C:\Users\Simon Okello\Documents\ANDELA\Flask\app\__init__.py:45]
2018-12-30 16:07:12,748 INFO: Flask_Demo startup [in C:\Users\Simon Okello\Documents\ANDELA\Flask\app\__init__.py:45]
2018-12-30 16:07:14,947 ERROR: Exception on / [GET] [in c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\app.py:1761]
Traceback (most recent call last):
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\_compat.py", line 35, in reraise
    raise value
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\Simon Okello\Documents\ANDELA\Flask\app\routes.py", line 36, in index
    return render_template("index.html", title="Home", form=form, posts=posts)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\flask\templating.py", line 134, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\environment.py", line 869, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\environment.py", line 830, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\environment.py", line 804, in _load_template
    template = self.loader.load(self, name, globals)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\loaders.py", line 125, in load
    code = environment.compile(source, name, filename)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\environment.py", line 591, in compile
    self.handle_exception(exc_info, source_hint=source_hint)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\Simon Okello\Documents\ANDELA\Flask\app\templates\index.html", line 19, in template
    {% endfor %}
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\environment.py", line 497, in _parse
    return Parser(self, source, name, encode_filename(filename)).parse()
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 901, in parse
    result = nodes.Template(self.subparse(), lineno=1)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 130, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 268, in parse_block
    node.body = self.parse_statements(('name:endblock',), drop_needle=True)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 165, in parse_statements
    result = self.subparse(end_tokens)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 130, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 212, in parse_if
    node.body = self.parse_statements(('name:elif', 'name:else',
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 165, in parse_statements
    result = self.subparse(end_tokens)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 144, in parse_statement
    self.fail_unknown_tag(token.value, token.lineno)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 97, in fail_unknown_tag
    return self._fail_ut_eof(name, self._end_token_stack, lineno)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 90, in _fail_ut_eof
    self.fail(' '.join(message), lineno)
  File "c:\users\simon okello\documents\andela\flask\env\lib\site-packages\jinja2\parser.py", line 59, in fail
    raise exc(msg, lineno, self.name, self.filename)
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'endfor'. Jinja was looking for the following tags: 'elif' or 'else' or 'endif'. The innermost block that needs to be closed is 'if'.
2018-12-30 16:08:18,818 INFO: Flask_Demo startup [in C:\Users\Simon Okello\Documents\ANDELA\Flask\app\__init__.py:45]
